core/domain/state_domain.py

```diff
@@ -1925,6 +1928,341 @@ def validate(self) -> None:
+                'Image tag does not have \'filepath-with-value\' attribute.'
+            )
+
+    for tag in soup.find_all('oppia-noninteractive-skillreview'):
+
+        if not tag.has_attr('text-with-value'):
+            raise utils.ValidationError(
+                'SkillReview tag does not have \'text-with-value\' '
+                'attribute.'
+            )
+
+        if tag['text-with-value'] is None:
+            raise utils.ValidationError(
+                'SkillReview tag should not have \'text-with-value\' '
+                'attribute value as None.'
+            )
+
+        if (
+            tag['text-with-value'].strip() in
+            ('&quot;&quot;', '', '\'\'', '\"\"')
+        ):
+            raise utils.ValidationError(
+                'SkillReview tag should not have \'text-with-value\' '
+                'attribute value as empty string.'
+            )
+
+        if not tag.has_attr('skill_id-with-value'):
+            raise utils.ValidationError(
+                'SkillReview tag does not have \'skill_id-with-value\' '
+                'attribute.'
+            )
+
+        if tag['skill_id-with-value'] is None:
+            raise utils.ValidationError(
+                'SkillReview tag should not have '
+                '\'skill_id-with-value\' attribute value as None.'
+            )
+
+        if (
+            tag['skill_id-with-value'].strip() in
+            ('&quot;&quot;', '', '\'\'', '\"\"')
+        ):
+            raise utils.ValidationError(
+                'SkillReview tag should not have '
+                '\'skill_id-with-value\' attribute value as empty.'
+            )
+
+    for tag in soup.find_all('oppia-noninteractive-video'):
+        if not tag.has_attr('start-with-value'):
+            raise utils.ValidationError(
+                'Video tag does not have \'start-with-value\' '
+                'attribute.'
+            )
+
+        if (
+            tag['start-with-value'].strip() in
+            ('&quot;&quot;', '', '\'\'', '\"\"')
+        ):
+            raise utils.ValidationError(
+                'Video tag \'start-with-value\' attribute should not'
+                'be empty.'
+            )
+
+        if not tag.has_attr('end-with-value'):
+            raise utils.ValidationError(
+                'Video tag does not have \'end-with-value\' '
+                'attribute.'
+            )
+
+        if (
+            tag['end-with-value'].strip() in
+            ('&quot;&quot;', '', '\'\'', '\"\"')
+        ):
+            raise utils.ValidationError(
+                'Video tag \'end-with-value\' attribute should not'
+                'be empty.'
+            )
+
+        start_value = tag['start-with-value'].strip()
+        end_value = tag['end-with-value'].strip()
+        if start_value > end_value and start_value != '0' and end_value != '0':
+            raise utils.ValidationError(
+                'Start value should not be greater than End value in Video tag.'
+            )
```

Comment:
This will work most of the time, but not all the time. You're comparing two strings here (which is okay in most cases). You should compare float values instead.

Reply:
Updated